Unit tests fail

The unit tests are randomly failing with 

```
--- FAIL: TestUpdateControlPlane (0.00s)
    --- FAIL: TestUpdateControlPlane/change_log_level (0.00s)
        config_updater_test.go:99: 
            Expected
                <chan string | len:1, cap:1>: 0xc00030a8c0
            to be empty
Running Suite: Static Suite - /home/runner/work/nginx-gateway-fabric/nginx-gateway-fabric/internal/mode/static
==============================================================================================================
Random Seed: 1727711008
```

Examples of failure:
- https://github.com/nginxinc/nginx-gateway-fabric/actions/runs/11109508301/job/30865132251?pr=2621#step:4:31
- https://github.com/nginxinc/nginx-gateway-fabric/actions/runs/11099159758/job/30832982203#step:4:31

--- a/internal/mode/static/config_updater_test.go
+++ b/internal/mode/static/config_updater_test.go
@@ -34,7 +34,6 @@ func TestUpdateControlPlane(t *testing.T) {
 	}
 
 	logger := zap.New()
-	fakeEventRecorder := record.NewFakeRecorder(1)
 	nsname := types.NamespacedName{Namespace: "test", Name: "test"}
 
 	tests := []struct {
@@ -81,6 +80,8 @@ func TestUpdateControlPlane(t *testing.T) {
 					return test.setLevelErr
 				},
 			}
+
+			fakeEventRecorder := record.NewFakeRecorder(1)
 
 			err := updateControlPlane(test.nginxGateway, logger, fakeEventRecorder, nsname, fakeLogSetter)
 

